paddle/framework/executor.h

```diff
@@ -40,6 +40,16 @@ class DeviceContextPool {
     return *pool;
   }
 
+  const platform::DeviceContext* Borrow(const platform::Place& place) {
+    auto range = device_contexts_.equal_range(place);
+    if (range.first == range.second) {
+      PADDLE_THROW(
+          "'Place' is not supported, Please re-compile with WITH_GPU "
+          "option");
```

Comment:
> "..., Please re-compile with WITH_GPU option"

This hint may not be intuitive. Currently `platform::Place` contains `CUDNNPlace, GPUPlace, CPUPlace, MKLDNNPlace`.
I think we can output this unsupported place by calling [this](https://github.com/PaddlePaddle/Paddle/blob/298dc8958d19867d0d2da1113f5d0ce7f7d618ed/paddle/platform/place.cc#L22).

Reply:
You're right. 
But we will remove the MKLPlace, CUDNNPlace soon. Place only represent address space, we will split it. 
 https://github.com/PaddlePaddle/Paddle/issues/6746
And, some interface of places isn't supported well. Such as is_cpu_place, default_gpu, we will not fix them because these things will be removed soon.

Reply:
People who look at this code may get confused. And you should add a TODO inside the code.

Reply:
I am working on removing the Place related problem.